docs/docs/creating-and-modifying-pages.md

```diff
@@ -92,13 +92,103 @@ exports.createPages = async ({ graphql, actions, reporter }) => {
       context: {
-        pagePath: path,
-      },
+        pagePath: path
+      }
     })
   })
 }
 ```
 
+## Trade-offs of querying for all fields in the context object of `gatsby-node.js`
+
+Imagine a scenario where you could query for all the parameters your template would need in the `gatsby-node.js`. What are the performance advantages this method might propose? In this section, you will look into this.
+
+In the initial approach you have seen how the `gatsby-node.js` file would have a query block like so :
+
+```graphql
+  const queryResults = await graphql(`
+    query AllProducts {
+      AllProducts {
+        nodes {
+          id
+        }
+      }
+    }
+  `);
+```
+
+Using the `id` as an access point to query for other properties in the template is the default approach. However, suppose you had a list of products with properties you would like to query for. Handling the query entirely from `gatsby-node.js`  would result in the query looking like this:
+
+```javascript:title=gatsby-node.js
+
+exports.createPages = async ({ graphql, actions }) => {
+  const { createPage } = actions;
+  const queryResults = await graphql(`
+    query AllProducts {
+      AllProducts {
+        nodes {
+          id
+          name
+          price
+          description
+        }
+      }
+    }
+  `);
+
+  const productTemplate = path.resolve(`src/templates/product.js`);
+  queryResults.data.allProducts.nodes.forEach(node => {
+    createPage({
+      path: `/products/${node.id}`,
+      component: productTemplate,
+      context: {
+        // This time the entire product is passed down as context
+        product
+      }
+    });
+  });
+};
+};
+```
+
+> You are now requesting all the data you need in a single query (this requires server-side support to fetch many products in a single database query).
+
+> As long as you can pass this data down to the template component via `pageContext`, there is no need for the template to make a GraphQL query at all.
+
+Your template file would look like this:
+
+```javascript:title=src/templates/product.js
+function Product({ pageContext }) {
+  return (
+    <div>
+      Name: {pageContext.name}
+      Price: {pageContext.price}
+      Description: {pageContext.description}
+    </div>
+  )
+}
+```
+
+### Performance implications
+
+Using the `pageContext` props in the template component can come with its performance advantages, of getting in all the data you need at build time; from the createPages API. This removes the need to have a GraphQL query in the template component.
+
+It does come with the advantage of querying your data from one place after declaring the `context` parameter.
+
+However, it doesn’t give you the opportunity to know what exactly you are querying for in the template and if any changes occur in the component query structure in gatsby-node.js. Hot reload is taken off the table and the site needs to be rebuilt for changes to reflect.
```

Comment:
```suggestion
However, it doesn’t give you the opportunity to know what exactly you are querying for in the template and if any changes occur in the component query structure in `gatsby-node.js`. [Hot reload](/docs/glossary#hot-module-replacement) is taken off the table and the site needs to be rebuilt for changes to reflect.
```

This is an opportunity to link to this word in the glossary 🙂

Reply:
hahaa, Thanks !